src/components/AnonymousChat/child/ChatItemTargetText.tsx

```diff
@@ -1,100 +1,121 @@
+  const isShowUserInfo = !isContinuous || isReplyPrompt || isReply;
+
+  const animatedBubbleStyle = useAnimatedStyle(() => ({
+    transform: [{translateX: bubblePosition.value}]
+  }));
+
+  const contextMenuActions = [
+    {title: 'Reply', systemIcon: 'arrow.turn.up.left'},
+    {title: 'Copy Message', systemIcon: 'square.on.square'}
+  ];
+
   const renderAvatar = React.useCallback(() => {
-    if (isContinuous) return <View style={styles.avatar} />;
+    if (!isShowUserInfo) return <View style={styles.avatar} />;
     return <View style={styles.mr8}>{avatar}</View>;
   }, []);
 
+  const onSwipeToReply = (direction) => {
+    if (direction === 'right') return;
+    if (swipeableRef.current) swipeableRef.current?.close();
+    setReplyPreview({
+      username: 'Anonymous User',
+      time: '4h',
+      message,
+      messageId: 'sdfs',
+      chatType: '',
+      messageType: 'regular'
+    });
+  };
+
   return (
-    <View style={styles.chatContainer}>
-      {renderAvatar()}
-      <View style={styles.textContainer}>
-        {!isContinuous && (
-          <View testID="chat-item-user-info" style={styles.chatTitleContainer}>
-            <Text style={styles.userText}>{username}</Text>
-            <View style={styles.dot} />
-            <Text style={styles.timeText}>{time}</Text>
-          </View>
-        )}
-        <Text testID="chat-item-message" style={styles.text}>
-          {message}
-        </Text>
+    <Swipeable
+      testID="swipeable"
+      ref={swipeableRef}
+      friction={3}
+      overshootFriction={2}
+      onSwipeableOpen={onSwipeToReply}
+      renderLeftActions={replyIcon}>
+      <View style={containerStyle(false, isReplyPrompt)}>
+        <Animated.View style={[styles.wrapper, animatedBubbleStyle]}>
+          {renderAvatar()}
+
+          <ContextMenu
+            previewBackgroundColor="transparent"
+            style={{flex: 1}}
+            actions={contextMenuActions}
+            onPress={(e) => onContextMenuPressed(e, data?.id, message)}>
+            <View style={styles.radius8}>
+              <View style={textContainerStyle(false)}>
+                {isShowUserInfo && (
+                  <View testID="chat-item-user-info" style={styles.chatTitleContainer}>
+                    <Text style={[styles.userText, textStyle(false)]}>{username}</Text>
+                    <View style={dotStyle(false)} />
+                    <Text style={[styles.timeText, textStyle(false)]}>{time}</Text>
+                  </View>
+                )}
+                <ChatReplyView
+                  type={chatType}
+                  messageType={messageType}
+                  replyData={data?.reply_data}
+                />
+                <Text testID="chat-item-message" style={messageStyle(false, messageType)}>
+                  {message}
+                </Text>
+              </View>
+            </View>
+          </ContextMenu>
+        </Animated.View>
       </View>
-    </View>
+    </Swipeable>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `Swipeable` component is missing a `renderRightActions` prop. If you want to allow swiping in both directions, you should provide this prop. If you only want to allow swiping to the left, then this is fine as is.



<!-- This is an auto-generated comment by OSS CodeRabbit -->